core/base/document.php

```diff
@@ -899,6 +899,20 @@ public function delete() {
 	 * @param array $elements
 	 */
 	protected function save_elements( $elements ) {
+		if ( ! current_user_can( 'unfiltered_html' ) ) {
+			$elements = Plugin::$instance->db->iterate_data( $elements, function( $element ) {
+				$element_instance = Plugin::$instance->elements_manager->create_element_instance( $element );
+				$element_controls = $element_instance->get_controls();
+				foreach ( $element_controls as $name => $config ) {
+					if ( Controls_Manager::URL === $config['type'] && isset( $element['settings'][ $name ] ) ) {
+						$element['settings'][ $name ]['url'] = (string) wp_http_validate_url( $element['settings'][ $name ]['url'] );
+					}
+				}
+
+				return $element;
+			} );
+		}
+
```

Comment:
This might cause a huge performance leak, and of course, if you do this check you don't need the second check inside `add_link_attributes` method